docs/build-notifications.rst

```diff
@@ -20,31 +20,256 @@ Take these steps to enable build notifications using email:
+
+Variable substitutions reference
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``{{event}}``
+  Event that triggered the webhook, one of ``build:triggered``, ``build:failed``, or ``build:passed``.
+
+``{{build.id}}``
+  Build ID.
+
+``{{build.commit}}``
+  Commit corresponding to the build, if present (otherwise ``""``).
+
+``{{build.url}}``
+  URL of the build.
+
+``{{build.docsurl}}``
+  URL of the documentation corresponding to the build.
+
+``{{organization.name}}``
+  Organization name (Commercial only).
+
+``{{organization.slug}}``
+  Organization slug (Commercial only).
+
+``{{project.slug}}``
+  Project slug.
+
+``{{project.name}}``
+  Project name.
+
+``{{project.url}}``
+  URL of the project :term:`dashboard`.
+
+``{{version.slug}}``
+  Version slug.
+
+``{{version.name}}``
+  Version name.
+
+Validating the payload
+~~~~~~~~~~~~~~~~~~~~~~
+
+After you add a new webhook, Read the Docs will generate a secret key for it
+and uses it to generate a hash signature (HMAC-SHA256) for each payload
+that is included in the ``X-Hub-Signature`` header of the request.
+
+.. figure:: /_static/images/webhooks-secret.png
+   :width: 80%
+   :align: center
+   :alt: Webhook secret
+
+   Webhook secret
+
+We highly recommend using this signature
+to verify that the webhook is coming from Read the Docs.
+To do so, you can add some custom code on your server,
+like this:
+
+.. code-block:: python
+
+   import hashlib
+   import hmac
+   import os
+
+
+   def verify_signature(payload, request_headers):
+       """
+       Verify that the signature of payload is the same as the one coming from request_headers.
+       """
+       digest = hmac.new(
+           key=os.environ["WEBHOOK_SECRET"].encode(),
+           msg=payload.encode(),
+           digestmod=hashlib.sha256,
+       )
+       expected_signature = digest.hexdigest()
+
+       return hmac.compare_digest(
+           request_headers["X-Hub-Signature"].encode(),
+           expected_signature.encode(),
+       )
+
+Legacy webhooks
+~~~~~~~~~~~~~~~
+
+Webhooks created before the custom payloads functionality was added to Read the Docs
+send a payload with the following structure:
+
+.. code-block:: json
+
+   {
+       "name": "Read the Docs",
+       "slug": "rtd",
+       "build": {
+           "id": 6321373,
+           "commit": "e8dd17a3f1627dd206d721e4be08ae6766fda40",
+           "state": "finished",
+           "success": false,
+           "date": "2017-02-15 20:35:54"
        }
+   }
+
+To migrate to the new webhooks and keep a similar structure,
+you can use this payload:
 
-You should now get notified on your webhook when your builds start and finish (failure/success)!
+.. code-block:: json
+
+   {
+       "name": "{{project.name}}",
+       "slug": "{{project.slug}}",
+       "build": {
+           "id": "{{build.id}}",
+           "commit": "{{build.commit}}",
+           "state": "{{build.state}}",
+           "success": "{{build.success}}",
+           "date": "{{build.date}}"
```

Comment:
these variables don't exist, we have `event` and `build.start_date`

Reply:
Corrected 👍🏽